electrode_models/metal_air_discret.py

```diff
@@ -0,0 +1,476 @@
+                resid[SVptr['phi_ed'][j]] =  i_io_int - i_io + i_el_int - i_el 
+            elif params['boundary'] == 'potential':                  
+                resid[SVptr['phi_ed'][j]] = (SV_loc[SVptr['phi_ed']] 
+                    - params['potential']) 
+        
+        # Calculate available surface area (m2 interface per m3 electrode):
+        A_avail = self.A_init - eps_product/self.th_oxide
+        # Convert to m2 interface per m2 geometric area:
+        A_surf_ratio = A_avail*self.dy
+        # Multiplier to scale phase destruction rates.  As eps_product drops 
+        # below the user-specified minimum, any reactions that consume the 
+        # phase have their rates quickly go to zero:
+        mult = tanh(eps_product / self.product_phase_min)
+
+        # Chemical production rate of the product phase: (mol/m2 interface/s)
+        sdot_product = (self.surf_obj.get_creation_rates(self.product_obj)
+            - mult * self.surf_obj.get_destruction_rates(self.product_obj))
+      
+        # Rate of change of the product phase volume fraction:
+        resid[SVptr['eps_product'][j]] = (SVdot_loc[SVptr['eps_product'][j]] 
+            - A_avail * np.dot(sdot_product, self.product_obj.partial_molar_volumes))
+
+        # Production rate of the electron (moles / m2 interface / s)
+        sdot_electron = (mult * self.surf_obj.get_creation_rates(self.host_obj)
+            - self.surf_obj.get_destruction_rates(self.host_obj))
+
+        # Positive Faradaic current corresponds to positive charge created in 
+        # the electrode:
+        i_Far = -(ct.faraday * sdot_electron)
+
+        # Double layer current has the same sign as i_Far
+        i_dl = self.i_ext_flag*(i_io_int - i_io)/A_surf_ratio - i_Far
+        resid[SVptr['phi_dl'][j]] = SVdot_loc[SVptr['phi_dl'][j]] - i_dl*self.C_dl_Inv
+        #change in concentration
+        sdot_elyte_host = (mult*self.surf_obj.get_creation_rates(self.elyte_obj)
+            - self.surf_obj.get_destruction_rates(self.elyte_obj))
+        sdot_elyte_host[self.index_Li] -= i_dl / ct.faraday 
+
+        resid[SVptr['C_k_elyte'][j]] = (SVdot_loc[SVptr['C_k_elyte'][j]] 
+            - (N_k_int - N_k + sdot_elyte_host * A_surf_ratio) 
+            * self.dyInv)/eps_elyte
+        j = 2
+        phi_ed = phi_ed_next
+        phi_elyte = phi_elyte_next
+        # print('phi_elyte = ', phi_elyte)
+        c_k_elyte = c_k_elyte_next
+        eps_product = eps_product_next
+
+        i_el_int = i_el
+
+         # Set Cantera object properties:
+        self.host_obj.electric_potential = phi_ed
+        self.elyte_obj.electric_potential = phi_elyte
+        self.elyte_obj.X = c_k_elyte
+        # Set microstructure multiplier for effective diffusivities
+        eps_elyte = eps_elyte_next
+        # Molar production rate of electrolyte species at the electrolyte-air 
+        # interface (kmol / m2 of interface / s) 
+        if self.name=='anode':
```

Comment:
Just to be 100% clear, we need to keep the `if/else` logic, here, since we are at the current collector boundary.